lab-tara/app/src/main/java/com/example/chatroomstatus/SignInActivity.java

```diff
@@ -0,0 +1,33 @@
+package com.example.chatroomstatus;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.EditText;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class SignInActivity extends AppCompatActivity {
+    @BindView(R.id.usernameInput) EditText mUsernameInput;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_sign_in);
+
+        ButterKnife.bind(this);
+    }
+
+    @OnClick(R.id.signin)
+    public void signin() {
+        String usernmame = mUsernameInput.getText().toString();
+
+        Intent intent = new Intent(this, StatusActivity.class);
+        intent.putExtra("username", usernmame);
```

Comment:
Consider defining "username" in one place to prevent issues with typos/mismatches.